### DISCUSSION TITLE
`cabal` file not found in dependency override

### DISCUSSION THREAD

**Discussion by peterbecich** [srid/haskell-flake#188] [Q&A] [answered]
I'm attempting to override the Hackage package `attoparsec-aeson`: 
https://hackage.haskell.org/package/attoparsec-aeson
https://github.com/haskell/aeson/tree/master/attoparsec-aeson

It is not available in NixPkgs: https://github.com/NixOS/nixpkgs/issues/247406

Tried two ways:
1 . `attoparsec-aeson.source = "2.2.0.0";` results in:

```
       > tar: */attoparsec-aeson/2.2.0.0/attoparsec-aeson.json: 
       > tar: */attoparsec-aeson/2.2.0.0/attoparsec-aeson.cabal:
```

Howover, the `.cabal` file is definitely in the archive, here it is in the `tar.gz` downloaded from Hackage:
```
$ ls
LICENSE  attoparsec-aeson.cabal  src
```

2.
Manual Flake input. (`attoparsec-aeson` lives in subdirectory of GitHub repo `haskell/aeson`.)
```
    attoparsec-aeson = {
      url = "github:haskell/aeson/274b562695863dc4a1b2e9a8c9d174c2b78cdd12?dir=attoparsec-aeson";
      flake = false;
    };

```
and 
```
attoparsec-aeson = inputs.attoparsec-aeson;
```

results in:

```
error: The option `lastModified' does not exist. Definition values:
```

Thanks for any help

Flake:
https://github.com/peterbecich/hackage-server/blob/update-flake/flake.nix

**Comment by srid** [comment, marked as answer]
For (1) you can try updating `nixpkgs`. For (2), it should be `attoparsec-aeson.source = inputs.attoparsec-aeson`(note the `.source`).

**Reply by peterbecich** [reply]
Thanks, great catch, I was missing `source`

**Reply by lrworth** [reply]
> For (1) you can try updating nixpkgs

Why did you suggest this? Is it because a new nixpkgs might have this package or a different reason?
